cmake/platform.cmake

```diff
@@ -75,6 +75,8 @@ elseif(UNIX OR MINGW)
     append_if(MKLDNN_WERROR CMAKE_CCXX_FLAGS "-Werror")
     append(CMAKE_CCXX_FLAGS "-fvisibility=internal")
     append(CMAKE_CXX_FLAGS "-fvisibility-inlines-hidden")
+    append(CMAKE_CXX_FLAGS "-mavx512f")
```

Comment:
This definitely breaks build for systems without Intel AVX512 support and compilers that are unaware of this option.